samza-core/src/main/java/org/apache/samza/processor/StreamProcessor.java

```diff
@@ -179,27 +184,36 @@ public synchronized void stop() {
     boolean containerShutdownInvoked = false;
     if (container != null) {
       try {
-        LOGGER.info("Shutting down container " + container.toString() + " from StreamProcessor");
+        LOGGER.info("Shutting down the container: {} of stream processor: {}.", container, processorId);
         container.shutdown();
+        LOGGER.info("Waiting {} milliseconds for the container: {} to shutdown.", taskShutdownMs, container);
+        jcContainerShutdownLatch.await(taskShutdownMs, TimeUnit.MILLISECONDS);
```

Comment:
The jcContainerShutdownLatch is for the JC to wait on container shutdown. Are you sure we can reuse it here? If so, let's rename it.